### ISSUE TITLE
Evaluate river trail

### ISSUE THREAD

**Issue by bhack** [software-engineering-amsterdam/MLitB#1]
I think that you can take advantages from the new [Ecmascript proposal river trail](http://wiki.ecmascript.org/doku.php?id=strawman:data_parallelism&rev=1343676564) instead of making plan on a not clear browser  roadmap to support WebCL . Intel has already code on github at https://github.com/IntelLabs/RiverTrail/wiki

**Comment by bhack** [comment]
Take a look also at this benchmark https://github.com/Maratyszcza/blis-bench/blob/master/README.md and to this presentation http://www.cs.utexas.edu/users/flame/BLISRetreat2014/slides/marat-blis-retreat2014/

**Comment by bhack** [comment]
@karpathy what do you think?

**Comment by bhack** [comment]
Referencing https://github.com/node-forward/discussions/issues/1

**Comment by tedmeeds** [comment]
Thanks, bhack for your suggestions.  Yes, we would love to accelerate our code, refactor it, etc.  If we find students to continue working on MLitB, acceleration will be a top priority.  Ted.

**Comment by bhack** [comment]
Do you think that there could be some interaction between this and @karpathy [convneths](https://github.com/karpathy/convnetjs)? In an interview @karpathy claimed: "Lastly, here's a crazy idea: massively distributed Neural Network training (think FoldIt, or SETI@Home), except every client merely visits a URL and right away starts to contribute Javascript compute time by sending gradient updates to a central server. A few issues have to be addressed first in terms of the modeling: vanilla Neural Networks have dense interactions so they are difficult to parallelize and naive use of distributed optimization techniques is likely to pose problems with stale gradients." My team is working in the caffe community and we have done some PR there. We are very interested in this kind of crowd computing large scale models. I don't know if a js solution that use parameter server in some way like non js [peetum](http://petuum.github.io) or a fully p2p can scale better with faster convergence.

**Comment by tedmeeds** [comment]
Sure, definitely.  Can we take this discussion offline?  Ted

**Comment by bhack** [comment]
Yes of course. If you want i can send you a mail..

**Comment by tedmeeds** [comment]
Thought you had it.  tmeeds@gmail.com.  Thanks.

**Comment by bjacob** [comment]
Don't hold your breath for widespread browser support for either WebCL (I don't know any browser vendor having plans to support it) or RiverTrail (I'm not aware of any browser vendor other than Mozilla showing an interest in it?)... instead, if you want something that runs in more than one browser, see how much you can do with JavaScript: Emscripten'ing C/C++ code into fast JavaScript, using Web Workers for multithreading, and if you need more power, look at SIMD.js and/or ongoing discussions to bring some pthreads-like multithreading capabilities to the Web.

**Comment by bhack** [comment]
There is an updated stage status at https://github.com/tc39/ecma262
